docs/user/dql/basics.rst

```diff
@@ -363,6 +363,17 @@ SQL query::
 	{
 	  "query" : "SELECT account_number FROM accounts/account"
 	}
+Example 4: Selecting From Index using Partition Shard
+-----------------------------------------------------------
+
+You can also specify a specific shard or partition to target using a routing hash key in ``PARTITION``.  You can target multiple shards by providing a list separated by commas.
+
+SQL query::
+
+	POST /_plugins/_sql
+	{
+	  "query" : "SELECT account_number FROM account PARTITION(shard1, shard2)"
```

Comment:
It may be confusing to use `PARTITION` an existing operation in SQL for this.

What about introducing a general notion of data source options, keeping them as key-value pairs in Relation that get analyzed by each StorageEngine. 

This would allow other data sources to benefit from this and avoid question about features that only apply to OpenSearch.

Reply:
This is exactly what the PARTITION in SQL is designed for.  Also, the proposal in the issue suggested using this syntax.  

I DO like your idea, but I'd rather stick with well-understood SQL syntax than introduce something new.